app/src/main/java/com/example/ohthmhyh/CustomAdapterHF.java

```diff
@@ -85,8 +87,22 @@ public void onItemMoved(int fromPosition, int toPosition) {
     @Override
     public void onItemSwiped(int position) {
         //TODO: Add confirmation alert dialog
-        habitList.removeHabit(position);
-        notifyItemRemoved(position);
+        DatabaseAdapter databaseAdapter;
+        databaseAdapter = new DatabaseAdapter();
+        databaseAdapter.pullHabitEvents(new DatabaseAdapter.HabitEventCallback() {
+            @Override
+            public void onHabitEventCallback(HabitEventList habitEvents) {
+                HabitEventList habitEventList;
+                habitEventList = habitEvents;
+                for (int index=0;index<habitEventList.size();index++){
+                    if (habitList.getHabit(position).getUHID()==habitEventList.getHabitEvent(index).getHabitUHID()){
+                        habitEventList.removeHabitEvent(index);
+                    }}
+                habitList.removeHabit(position);
+                notifyItemRemoved(position);
```

Comment:
```suggestion
                // TODO: Look to add this as a method to HabitEventList.
                for (int index = 0; index < habitEvents.size(); index++){
                    if (habitList.getHabit(position).getUHID() == habitEventList.getHabitEvent(index).getHabitUHID()) {
                        habitEventList.removeHabitEvent(index);
                    }
                }
                habitList.removeHabit(position);
                notifyItemRemoved(position);
```

Reply:
This change makes it a bit easier to read. If you agree with these changes and submit them, please test that these changes don't break anything.